testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/elytron/CustomCredentialSecurityFactoryTestCase.java

```diff
@@ -0,0 +1,200 @@
+            cli.sendLine(String.format(
+                    "/core-service=management/management-interface=http-interface:undefine-attribute(name=http-authentication-factory)",
+                    MANAGEMENT_FILESYSTEM_NAME));
+            cli.sendLine(String.format(
+                    "/subsystem=elytron/http-authentication-factory=%s:remove()",
+                    MANAGEMENT_FILESYSTEM_NAME, PREDEFINED_HTTP_SERVER_MECHANISM_FACTORY));
+            cli.sendLine(String.format("/subsystem=elytron/security-domain=%s:remove()", MANAGEMENT_FILESYSTEM_NAME));
+            cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s:remove()", MANAGEMENT_FILESYSTEM_NAME));
+        }
+        FileUtils.deleteDirectory(tempFolder);
+    }
+
+    @BeforeClass
+    public static void setupServer() throws Exception {
+        CONTROLLER.start();
+        host = TestSuiteEnvironment.getServerAddress();
+        prepareServerConfiguration();
+    }
+
+    @AfterClass
+    public static void resetServer() throws Exception {
+        try {
+            resetServerConfiguration();
+        } finally {
+            CONTROLLER.stop();
+        }
+    }
+
+    /**
+     * Test whether existing user with correct password has granted access through http-interface secured by Elytron
+     * http-authentication-factory.
+     */
+    @Test
+    public void testCorrectUser() throws Exception {
+        try (CLIWrapper cli = new CLIWrapper(true)) {
+            StringBuilder configuration = new StringBuilder("throwException").append("=").append(false);
+
+            cli.sendLine(
+                String.format(
+                    "/subsystem=elytron/custom-credential-security-factory=%s:write-attribute(name=configuration, value={%s})",
+                    CUSTOM_CRED_SEC_FACTORY_NAME, configuration));
+            cli.sendLine("reload");
+        }
+        CoreUtils.makeCallWithBasicAuthn(createSimpleManagementOperationUrl(), USER, CORRECT_PASSWORD, SC_OK);
+    }
+
+    /**
+     * Test where is thrown exception in custom-credential-security-factory.
+     */
+    @Test
+    public void testCorrectUserCustomCredentialSecurityFactoryException() throws Exception {
+        try (CLIWrapper cli = new CLIWrapper(true)) {
+            StringBuilder configuration = new StringBuilder("throwException").append("=").append(true);
+
+            cli.sendLine(String.format(
+                "/subsystem=elytron/custom-credential-security-factory=%s:write-attribute(name=configuration, value={%s})",
+                CUSTOM_CRED_SEC_FACTORY_NAME, configuration));
+            cli.sendLine("reload");
+        }
+        CoreUtils.makeCallWithBasicAuthn(createSimpleManagementOperationUrl(), USER, CORRECT_PASSWORD,
+            HttpStatus.SC_INTERNAL_SERVER_ERROR);
+    }
+
+    private URL createSimpleManagementOperationUrl() throws URISyntaxException, IOException {
+        return new URL("http://" + host + ":9990/management?operation=attribute&name=server-state");
```

Comment:
This should use the `TestSuiteEnvironment.getServerPort()` instead of the hard-coded port.